src/server/scripts/Shadowlands/Sanctum of Domination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,3702 @@
+                    if (jaina->IsAIEnabled())
+                        jaina->AI()->DoAction(ACTION_OPEN_PORTAL_TO_PHASE_THREE);
+                }
+
+                break;
+            }
+
+            case ACTION_INITIATE_PHASE_THREE:
+            {
+                me->NearTeleportTo(SylvanasPhase3Pre, false);
+
+                if (Creature* shadowCopy = ObjectAccessor::GetCreature(*me, _shadowCopyGUID[0]))
+                    shadowCopy->NearTeleportTo(me->GetPosition(), false);
+
+                if (Creature* shadowCopy = ObjectAccessor::GetCreature(*me, _shadowCopyGUID[1]))
+                    shadowCopy->NearTeleportTo(me->GetPosition(), false);
+
+                if (Creature* shadowCopy = ObjectAccessor::GetCreature(*me, _shadowCopyGUID[2]))
+                    shadowCopy->NearTeleportTo(me->GetPosition(), false);
+
+                if (Creature* shadowCopy = ObjectAccessor::GetCreature(*me, _shadowCopyGUID[3]))
+                    shadowCopy->NearTeleportTo(me->GetPosition(), false);
+
+                DoCastSelf(SPELL_RANGER_BOW_STANCE, true);
+                break;
+            }
+
+            case ACTION_START_PHASE_THREE:
+            {
+                DoCastSelf(SPELL_RIVE_DISAPPEAR, true);
+
+                if (Creature* shadowCopy = ObjectAccessor::GetCreature(*me, _shadowCopyGUID[0]))
+                {
+                    me->SetNameplateAttachToGUID(_shadowCopyGUID[0]);
+
+                    shadowCopy->CastSpell(SylvanasPlaform1Pos, SPELL_DOMINATION_CHAINS_JUMP, true);
+
+                    me->SendPlayOrphanSpellVisual(SylvanasPlaform1Pos, SPELL_VISUAL_WINDRUNNER_01, 0.5f, true, false);
+                }
+
+                scheduler.Schedule(250ms, [this](TaskContext context)
+                {
+                    Talk(SAY_START_PHASE_THREE);
+                });
+
+                scheduler.Schedule(525ms, [this](TaskContext context)
+                {
+                    me->NearTeleportTo(SylvanasPlaform1Pos, false);
+                });
+
+                scheduler.Schedule(1s, [this](TaskContext context)
+                {
+                    DoAction(ACTION_REACTIVATE_AI_AFTER_EVENT);
+
+                    DoCastSelf(SPELL_DOMINATION_CHAINS_SPIN, true);
+
+                    me->RemoveAura(SPELL_BANSHEE_SHROUD);
+                });
+
+                break;
+            }
+
+            default:
+                break;
+        }
+    }
+
+    void OnSuccessfulSpellCast(SpellInfo const* spell) override
+    {
+        switch (spell->Id)
+        {
+            case SPELL_RANGER_SHOT:
+            {
+                _lastSpellUsed = spell->Id;
+                break;
+            }
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        scheduler.Update(diff);
+
+        if (!UpdateVictim())
+            return;
+
+        events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        while (uint32 eventId = events.ExecuteEvent())
```

Comment:
Use the hook OnExecuteEvent from BossAI

Reply:
I used Lich King.cpp as reference and it used this one, so I'll keep it.